src/preferences/upgrade.cpp

```diff
@@ -66,6 +67,32 @@ WaveformWidgetType::Type upgradeToAllShaders(WaveformWidgetType::Type waveformTy
     }
     return WWT::AllShaderRGBWaveform;
 }
+
+VSyncThread::VSyncMode upgradeDeprecatedVSyncModes(int configVSyncMode) {
+    using VT = VSyncThread;
+    if (configVSyncMode >= 0 || configVSyncMode <= static_cast<int>(VT::ST_COUNT)) {
+        switch (static_cast<VSyncThread::VSyncMode>(configVSyncMode)) {
+        case VT::ST_DEFAULT:
+            return VT::ST_DEFAULT;
+        case VT::ST_MESA_VBLANK_MODE_1_DEPRECATED:
+            return VT::ST_DEFAULT;
+        case VT::ST_SGI_VIDEO_SYNC_DEPRECATED:
+            return VT::ST_DEFAULT;
+        case VT::ST_OML_SYNC_CONTROL_DEPRECATED:
+            return VT::ST_DEFAULT;
+        case VT::ST_FREE:
+            return VT::ST_FREE;
+        case VT::ST_TIMER:
+            return VT::ST_TIMER;
+        case VT::ST_PLL:
+            return VT::ST_PLL;
+        case VT::ST_COUNT:
```

Comment:
let's remove this case and then change the range check above to < instead of <=.  That's a little more clear.